apps/protailwind/src/pages/invoices/[merchantChargeId].tsx

```diff
@@ -3,91 +3,69 @@ import {DownloadIcon} from '@heroicons/react/outline'
 import {convertToSerializeForNextResponse} from '@skillrecordings/commerce-server'
 import {useLocalStorage} from 'react-use'
 import {GetServerSideProps} from 'next'
-import {stripe} from '@skillrecordings/commerce-server'
-import {Coupon, getSdk, MerchantProduct} from '@skillrecordings/database'
+import {Coupon, MerchantProduct} from '@skillrecordings/database'
 import {Stripe} from 'stripe'
 import fromUnixTime from 'date-fns/fromUnixTime'
 import Layout from 'components/layout'
 import format from 'date-fns/format'
-import {prisma} from '@skillrecordings/database'
 import {trpc} from '../../trpc/trpc.client'
 import {Transfer} from '../../purchase-transfer/purchase-transfer'
 import {MailIcon} from '@heroicons/react/solid'
+import {z} from 'zod'
+import {useRouter} from 'next/router'
 
-export const getServerSideProps: GetServerSideProps = async ({
-  res,
-  req,
-  query,
-}) => {
-  const {merchantChargeId} = query
-  const {getProduct, getPurchaseForStripeCharge} = getSdk()
-  if (merchantChargeId) {
-    const merchantCharge = await prisma.merchantCharge.findUnique({
-      where: {
-        id: merchantChargeId as string,
-      },
-      select: {
-        id: true,
-        identifier: true,
-        merchantProductId: true,
-      },
-    })
-
-    if (merchantCharge && merchantCharge.identifier) {
-      const charge = await stripe.charges.retrieve(merchantCharge.identifier, {
-        expand: ['customer'],
-      })
-
-      const purchase = await getPurchaseForStripeCharge(
-        merchantCharge.identifier,
-      )
-      const bulkCoupon = purchase && purchase.bulkCoupon
-
-      const product = await getProduct({
-        where: {id: purchase?.productId},
-      })
```

Comment:
rather than having to copy this logic into each app's invoice page, I moved it to a skill-lesson tRPC query

Reply:
this is great ⭐️